src/Compilers/CSharp/Test/Emit2/Semantics/InlineArrayTests.cs

```diff
@@ -20598,30 +20611,42 @@ .maxstack  3
             CompileAndVerify(comp, expectedOutput: " 0 1 2 3", verify: Verification.Fails).VerifyDiagnostics();
         }
 
-        [Fact]
+        [ConditionalFact(typeof(CoreClrOnly))]
         public void Foreach_InAsync_07()
         {
             var src = @"
 class Program
 {
-    static async void Test()
+    static Buffer4<int> s_buffer;
+
+    static async System.Threading.Tasks.Task Main()
     {
+        s_buffer[1] = 3;
+
         foreach (ref readonly int y in GetBuffer())
         {
+            System.Console.Write(y);
         }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Please add a test scenario with ```await System.Threading.Tasks.Task.Yield();``` in the loop.

Reply:
Perhaps two separate scenarios with an `await` at the beginning of the loop and an `await` at the end of the loop.